src/curl.cpp

```diff
@@ -2276,6 +2276,7 @@ void S3fsCurl::insertV4Headers(const string &op, const string &path, const strin
   requestHeaders = curl_slist_sort_insert(requestHeaders, "host", get_bucket_host().c_str());
   requestHeaders = curl_slist_sort_insert(requestHeaders, "x-amz-content-sha256", contentSHA256.c_str());
   requestHeaders = curl_slist_sort_insert(requestHeaders, "x-amz-date", date8601.c_str());
+  requestHeaders = curl_slist_sort_insert(requestHeaders, "x-amz-request-payer", "requester");
```

Comment:
Why do you add this header unconditionally?

Reply:
Because it fixed the issue I was having, and as I stated, it would ideally be added as an option

Reply:
Sending unneeded headers to some S3 implementations will cause requests to fail, as with S3Proxy running in Travis.  Please add a flag.

Reply:
Hi, I have created this [PR](https://github.com/s3fs-fuse/s3fs-fuse/pull/1199) implementing this flag as a new option. I hope this helps!